### ISSUE TITLE
Consider Moving Docs to Docusaurus

### ISSUE THREAD

**Issue by GrahamSH-LLK** [ScratchAddons/website-v2#283]
Docusaurus is a industry-standard open-source docs platform that makes really good docs. [Turbowarp](//docs.turbowarp.org) uses it. The fact that [ScratchTools](//docs.scratchtools.app) has better docs than us can be attributed to it too.
It still works with the github format, just instead of rolling our own docs we can use a proper docs setup. It comes with added benefits like search.

**Comment by WorldLanguages** [comment]
cc @Hans5958 

Search would be nice.
And I'm not convinced about having the user-facing FAQ and docs for developers all in one system...

**Comment by Hans5958** [comment]
MkDocs, Docusaurus, etc are good, but then you would either (a) rework it to have the same style, which is a little bit difficult, or (b) drop everything and have different style, at least we got the goodies of a docs.

Speaking about SPA, having Docusaurus would mean a bigger overhead, since it uses React, compared to non-SPA solutions like Hugo, MkDocs, which is still traditional.

Search can be implemented with a little bit of work (#168).

**Comment by GrahamSH-LLK** [comment]
Docusaurus and MKDocs are both great. Docusaurus feels a little more user friendly and MKDocs feels more like internal technical docs.

**Comment by WorldLanguages** [comment]
If we add search, we should restrict it to developer docs only, or somehow add a toggle for that. I hate it when I go to a Chrome extensions API doc, such as https://developer.chrome.com/docs/extensions/reference/storage/, then I try to use the search bar and it searches the whole Chrome docs, not just Chrome extensions.

**Comment by Hans5958** [comment]
Correct. This is the proof-of-concept made by @GrahamSH-LLK for this issue, though it seems it is pretty abandoned at this point.
